.circleci/config.yml

```diff
@@ -1224,22 +1260,36 @@ jobs:
           name: Prepare test environment (compile tests and cache results)
           working_directory: op-acceptance-tests
           command: go test -v ./... || true # ignore failures; we're just compiling the tests
-      # Run the actual acceptance tests with optimized settings
       - run:
-          name: Run acceptance tests (devnet=<<parameters.devnet>>, gate=<<parameters.gate>>)
+          name: Stop the job if the devnet failed to start
+          command: circleci-agent step halt
+          when: on_fail
```

Comment:
This command will not stop the rest of the workflow - just a particular job. And since the job is already failing (`when: on_fail`), there is no need to stop it.

Reply:
Yes, I want to stop the job and not the workflow. This command stops the execution of the subsequent steps in the job, which is what I want. A test confirmed this behaviour: https://app.circleci.com/pipelines/github/ethereum-optimism/optimism/88066/workflows/224e2b1d-1a1a-4cba-8695-c28888dda8c6/jobs/3467342

Reply:
I think I am missing something - in the workflow you linked, this step was never executed because `Start devnet` step exited with code 1. If a step exits with non-zero exit code, the subsequent steps will not be executed unless marked with `on_fail` or `always`.

What is the scenario that you are trying to address?

Reply:
There is a discord notification later in the job which is marked as `on_fail`. However, if the devnet fails I still don't want to run it (because this would send duplicate notifications).

Pseudo-code/flow:
```
- start devnet
- if devnet fails to start, notify Platforms Dev X and stop the job # <-- the step in question
- run acceptance tests
- output test logs/artifacts
- if tests fail, notify Platforms Dev X and/or Protocol
```

This is the "trick" I've chosen because it seems you can't conditionally run one step based on the exit code of another nicely.

Does that make sense?

Reply:
Resolving this comment as you've since approved the PR. Thanks!